### PR TITLE
change TFPark to from_train_op

### PR DESCRIPTION
**Pull request by yangw1234**
Hi @jason-dai, I still think from_train_op is the more intuitive API for user and it supports multiple optimizers. Maybe we can make `loss` optional if user does not have the access to `loss`.

### CODE REVIEW

**Comment by yangw1234** [comment]
http://172.168.2.101:8080/job/ZOO-PR-Validation-Spark-2.4/1383/ ut passed

**Comment by jason-dai** [comment]
Two questions:
1) Can we support any `train_op` (assuming the user has used `ZooOptimizer`)?
2) The API still require `loss`; then why is API is more convenient than asking the user to pass `loss` and `optimizer` to TFOptimizer?

File: docs/docs/ProgrammingGuide/TFPark/tensorflow.md [hunk 1]
```diff
@@ -206,26 +205,10 @@ with slim.arg_scope(lenet.lenet_arg_scope()):
...
-from tensorflow.keras.models import Model
-from tensorflow.keras.layers import *
-
-data = Input(shape=[28, 28, 1])
 
-x = Flatten()(data)
-x = Dense(64, activation='relu')(x)
-x = Dense(64, activation='relu')(x)
-predictions = Dense(10, activation='softmax')(x)
+optimizer = ZooOptimizer(tf.train.AdamOptimizer(1e-3))
```
Comment (jason-dai) [inline docs/docs/ProgrammingGuide/TFPark/tensorflow.md]: Maybe we can use something like `tfpark.Optim` or `tfpark.OptimMethod` instead of `ZooOptimizer`?

**Comment by yangw1234** [comment]
Hi @jason-dai I'll rethink about this. Close it for now.
